File: backend/app/schemas/transcript.py
from typing import List, Optional
from pydantic import BaseModel


class TranscriptSnippet(BaseModel):
    text: str
    start: float
    duration: float


class TranscriptResponse(BaseModel):
    video_id: str
    available: bool
    language: Optional[str] = None
    language_code: Optional[str] = None
    is_generated: Optional[bool] = None
    snippets: List[TranscriptSnippet] = []
    reason: Optional[str] = None

File: frontend/src/types/transcript.ts
export interface TranscriptSnippet {
    text: string
    start: number
    duration: number
}

export interface TranscriptResponse {
    video_id: string
    available: boolean
    language: string | null
    language_code: string | null
    is_generated: boolean | null
    snippets: TranscriptSnippet[]
    reason: string | null
}